app/models/hold.rb

```diff
@@ -54,7 +53,8 @@ def lift_hold
 
   def conflicting_ongoing_rental
     # it couldnt be possible that the same item has already been picked up twice under the same rental
-    Rental.picked_up.find_by('start_time <= :hold_end_time AND end_time >= :hold_start_time AND item_id = :hold_item_id',
-                             hold_start_time: start_time, hold_end_time: end_time, hold_item_id: item.id)
+    Rental.picked_up.joins('INNER JOIN rentals_items ON rentals_items.rental_id = rentals.id')
+          .find_by('start_time <= :hold_end_time AND end_time >= :hold_start_time AND rentals_items.item_id = :hold_item_id',
+                   hold_start_time: start_time, hold_end_time: end_time, hold_item_id: item.id)
```

Comment:
Where are these local variables coming from?

Reply:
instance variables

Reply:
Oh, they're attributes of the model. I got so confused, I forgot I was looking at an instance method in a model file.